jsonnet/components/prometheus.libsonnet

```diff
@@ -215,26 +215,24 @@ function(params)
 
     kubeRbacProxySecret: generateSecret.staticAuthSecret(cfg.namespace, cfg.commonLabels, 'kube-rbac-proxy'),
 
-    // this secret allows us to identify alerts that
-    // are coming from platform monitoring
-    additionalAlertManagerRelabellingSecret: {
-      apiVersion: 'v1',
-      kind: 'Secret',
+    // The default AlertRelabelConfig allows us to identify alerts that are coming
+    // from platform monitoring.
+    defaultAlertRelabelConfigs: {
+      apiVersion: 'monitoring.openshift.io/v1alpha1',
+      kind: 'AlertRelabelConfig',
       metadata: {
-        name: 'alert-relabel-configs',
+        name: 'zz99-default',
```

Comment:
I'm assuming that you've chose the `zz99-` prefix to ensure that it would be appended last in the final config. What if the relabel controller had this rule hardcoded. As far as I can tell, we went with a static YAML manifest because it needed no customization which isn't the case anymore with your PR.

Reply:
We can definitely do that. I actually wasn't sure if it should come first or last though.

Reply:
I'd say last since we want to enforce this label. Now it's also dependent on the discussion regarding `openshift_io_alert_source` :)

Reply:
True. Also, I guess we may have to use a different label, not just a different value anyway, because this would overwrite the static label added to user-defined alerts.

Reply:
I dropped all the static bits from the assets. The controller now defines the default config, and always appends it to the list. I've tested that it all seems to work even when the secret doesn't exist at Prometheus startup. The controller creates it shortly after, and it's automatically picked up.